applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -123,6 +125,7 @@ struct subscriber {
 	bool is_usb;
 	uint8_t report_max;
 	uint8_t report_cnt;
+	uint8_t kbd_leds;
```

Comment:
The report_data structure was created so we would avoid creating report specific members.
We could use report_data for output reports.

Reply:
Report data is used to handle HID input data and it seems tailored for that purpose. I think we cannot simply use it for HID output reports (it even has no member that could be used to remember bytes related to given HID output report).

Reply:
Also state of the keyboard LEDs must be remembered separately for every subscriber (or actually report_state).  The report_state has no place to hold the output report data.

Reply:
I tried to use the simplest possible approach to handle only HID keyboard LEDs output report in HID state (since it's tightly coupled with keyboard keys HID input report). This also helps to avoid increasing memory footprint.

Reply:
Other HID output reports could be handled in report-specific manner by dedicated modules (in case someone adds them). I am not sure if we can introduce a generic way of handling HID output reports in HID state. Many HID output reports may need to be handled in a specific way (e.g. HID output report related to configuration channel).

Reply:
> Report data is used to handle HID input data and it seems tailored for that purpose. 

Not really. It was added to handle data that's part of the report. This involves two types (in our case) absolute (buttons) and relative (axis). I don't see output report data as being any different here.

> Also state of the keyboard LEDs must be remembered separately for every subscriber (or actually report_state). The report_state has no place to hold the output report data.
Report state was not created to store any data but to track state of connection and subscription. I don't think data should go there. As for storing data per subscriber. The question is how much of it is a valid case (e.g. can we request host to pass the data again after switching subscribers) and if so does it validate placing data in structure not intended to hold it (maybe we should have a new generic linking for report data connected with output reports).